apps/backend/src/action/logic/remove-path-card.ts

```diff
@@ -1,27 +1,108 @@
+export type BoardCardEvent = BoardEvent | PathCardHasBeenRemovedEvent;
+
+const TargetCannotBeRemovedError = error("TargetCannotBeRemovedError");
+const RepositoryWriteError = error("RepositoryWriteError");
+export type RepositoryWriteError = ReturnType<typeof RepositoryWriteError>;
+export type TargetCannotBeRemovedError = ReturnType<
+  typeof TargetCannotBeRemovedError
+>;
+export type RemovePathCardError =
+  | TargetCannotBeRemovedError
+  | RepositoryWriteError
+  | RepositoryReadError;
 
 export interface RemovePathCard {
-  (source: EventSource<Event>, command: RockfallCommand): ResultAsync<
+  (source: EventSource<BoardCardEvent>, command: RockfallCommand): ResultAsync<
     PathCardHasBeenRemovedEvent,
-    Error
+    RemovePathCardError
   >;
 }
 
+const NonRemovablePathCard = Object.freeze([
+  PathCard.START,
+  PathCard.GOAL_COAL_BOTTOM_LEFT,
+  PathCard.GOAL_COAL_BOTTOM_RIGHT,
+  PathCard.GOAL_GOLD,
+]);
+
+const isRemovablePathCard = Either.fromPredicate<
+  Placement,
+  TargetCannotBeRemovedError
+>(
+  (placement) => !NonRemovablePathCard.includes(placement.card),
+  (placement) =>
+    TargetCannotBeRemovedError(
+      `the path card ${placement.card} cannot be removed`
+    )
+);
+
+const findPlacementByPosition =
+  (command: RockfallCommand) => (board: Placement[]) =>
+    Array.findFirst<Placement>(
+      flow(prop("position"), Vec.eq(command.data.position))
+    )(board);
+
+const checkPathCardHasBeenPlaced =
+  (command: RockfallCommand) => (board: Placement[]) =>
+    pipe(
+      board,
+      findPlacementByPosition(command),
+      Either.fromOption(() =>
+        TargetCannotBeRemovedError("empty placement cannot be removed")
+      ),
+      Either.chain(isRemovablePathCard),
+      Either.matchW(err, always(ok(command)))
+    );
+
+const appendEventToEventSource =
+  (source: EventSource<BoardCardEvent>, command: RockfallCommand) => () =>
+    pipe(
+      PathCardHasBeenRemovedEvent(command.data),
+      async (event) => {
+        await source.append(event);
+        return event;
+      },
+      (event) =>
+        ResultAsync.fromPromise(
+          event,
+          always(RepositoryWriteError("failed to write event to repository"))
+        )
+    );
+
 /**
  * *description*
  * remove path card on the board
  *
  * *param* source - event source
  * *param* command - rockfall command
  */
-export const removePathCard: RemovePathCard = (source, command) => {
+
+export const removePathCard: RemovePathCard = (source, command) =>
   //@todo: read from event source
-  //@todo: check path card has been placed
-  //@todo: append event to event source
-  return errAsync(new Error("not implemented"));
-};
+  /**
+   * @question
+   *  Would it be possible to include events for removing cards in the BoardEvent type as well,
+   *  so that getCurrentPlacements can obtain the correct PathCard on the board?
+   */
+  getCurrentPlacements(source as EventSource<BoardEvent>)
+    //@todo: check path card has been placed
```

Comment:
@JohnsonMao could you help me clear the todo comment after implementation finished, thx.